contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -4977,6 +4983,48 @@ rename_tsql_db(char *old_db_name, char *new_db_name)
 		CommitTransactionCommand();
 }
 
+/* 
+ * user_exists_for_db
+ *	returns true if the user/role exists in the sys.babelfish_authid_user_ext catalog,
+ *	false otherwise.
+ */
+bool
+user_exists_for_db(const char *db_name, const char *user_name)
+{
+	Relation		bbf_authid_user_ext_rel;
+	HeapTuple		tuple_user_ext;
+	ScanKeyData		key[2];
+	TableScanDesc		scan;
+	NameData		*rolname;
+	bool			user_exists = false;
+
+	bbf_authid_user_ext_rel = table_open(get_authid_user_ext_oid(),
+										 RowExclusiveLock);
+	rolname = (NameData *) palloc0(NAMEDATALEN);
+	snprintf(rolname->data, NAMEDATALEN, "%s", user_name);
+	ScanKeyInit(&key[0],
+				Anum_bbf_authid_user_ext_rolname,
+				BTEqualStrategyNumber, F_NAMEEQ,
+				NameGetDatum(rolname));
+	ScanKeyInit(&key[1],
+				Anum_bbf_authid_user_ext_database_name,
+				BTEqualStrategyNumber, F_TEXTEQ,
+				CStringGetTextDatum(db_name));
+	
+	scan = table_beginscan_catalog(bbf_authid_user_ext_rel, 2, key);
+
+	tuple_user_ext = heap_getnext(scan, ForwardScanDirection);
+
+	if (HeapTupleIsValid(tuple_user_ext))
+	{
+		user_exists = true;
+	}
+
+	table_endscan(scan);
+	table_close(bbf_authid_user_ext_rel, RowExclusiveLock);
+	return user_exists;
+}
+
```

Comment:
rolname is primary key.
Use syscache lookup using rolname & just compare the dbname.